### PR TITLE
upgrade rusty_v8 to 0.16.0

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

**Comment by ry** [comment]
I'm experiencing segfaults using rusty_v8 0.16.0.

On linux x64 
```
~/src/deno> lldb -- ./target/debug/deno test --coverage --unstable test_coverage.ts
(lldb) target create "./target/debug/deno"
Current executable set to './target/debug/deno' (x86_64).
(lldb) settings set -- target.run-args  "test" "--coverage" "--unstable" "test_coverage.ts"
(lldb) r
Process 21920 launched: './target/debug/deno' (x86_64)
Process 21920 stopped
* thread #1, name = 'deno', stop reason = signal SIGSEGV: address access protected (fault address: 0x55555c658e70)
    frame #0: 0x000055555c658e70
->  0x55555c658e70: addb   $0x0, 0x55555c3e(%rcx)
    0x55555c658e77: addb   %dl, %al
    0x55555c658e79: cmpl   $0x55555c, (%rsi)         ; imm = 0x55555C
    0x55555c658e7f: addb   %al, (%rax)
(lldb) up
frame #1: 0x000055555983fbdd deno`v8_inspector::V8InspectorImpl::V8InspectorImpl(v8::Isolate*, v8_inspector::V8InspectorClient*) at v8-inspector-impl.cc:65
(lldb) up
frame #2: 0x000055555983fb6f deno`v8_inspector::V8Inspector::create(v8::Isolate*, v8_inspector::V8InspectorClient*) at v8-inspector-impl.cc:54
(lldb) up
frame #3: 0x000055555904ec11 deno`::v8_inspector__V8Inspector__create(isolate=0x0000398600000000, client=0x000055555c658950) at binding.cc:1836
(lldb) bt
error: need to add support for DW_TAG_base_type '()' encoded with DW_ATE = 0x7, bit_size = 0
error: need to add support for DW_TAG_base_type '()' encoded with DW_ATE = 0x7, bit_size = 0
error: need to add support for DW_TAG_base_type '()' encoded with DW_ATE = 0x7, bit_size = 0
* thread #1, name = 'deno', stop reason = signal SIGSEGV: address access protected (fault address: 0x55555c658e70)
    frame #0: 0x000055555c658e70
    frame #1: 0x000055555983fbdd deno`v8_inspector::V8InspectorImpl::V8InspectorImpl(v8::Isolate*, v8_inspector::V8InspectorClient*) at v8-inspector-impl.cc:65
    frame #2: 0x000055555983fb6f deno`v8_inspector::V8Inspector::create(v8::Isolate*, v8_inspector::V8InspectorClient*) at v8-inspector-impl.cc:54
  * frame #3: 0x000055555904ec11 deno`::v8_inspector__V8Inspector__create(isolate=0x0000398600000000, client=0x000055555c658950) at binding.cc:1836
    frame #4: 0x0000555557a01ad1 deno`rusty_v8::inspector::V8Inspector::create::h11e3a071d2a56027(isolate=0x0000398600000000, client=0x000055555c658950) at inspector.rs:867
    frame #5: 0x000055555792bc71 deno`deno_runtime::inspector::DenoInspector::new::_$u7b$$u7b$closure$u7d$$u7d$::hbb58c30a9f306579(self_ptr=0x000055555c658950) at inspector.rs:498
    frame #6: 0x000055555792dd30 deno`deno_runtime::inspector::new_box_with::h8e0957ee30ea3367(new_fn=<unavailable>) at inspector.rs:1059
    frame #7: 0x000055555774e6b7 deno`deno_runtime::inspector::DenoInspector::new::h254804457ca2b643(js_runtime=0x00007fffffff2778, server=Option<alloc::sync::Arc<deno_runtime::inspector::InspectorServer>> @ 0x00007fffffff24a0) at inspector.rs:494
    frame #8: 0x00005555576e220c deno`deno_runtime::worker::MainWorker::from_options::hd12393fdf5aadad4(main_module=<unavailable>, permissions=<unavailable>, options=0x00007fffffff8780) at worker.rs:84
    frame #9: 0x0000555555bf8990 deno`deno::create_main_worker::hd5555bdb03580017(program_state=0x000055555c57eab0, main_module=<unavailable>, permissions=<unavailable>) at main.rs:209
    frame #10: 0x00005555564fc513 deno`deno::test_command::_$u7b$$u7b$closure$u7d$$u7d$::h821f74dbb689c171((null)=ResumeTy @ 0x00007fffffffb0c0) at main.rs:1032
    frame #11: 0x00005555563e1e9b deno`_$LT$core..future..from_generator..GenFuture$LT$T$GT$$u20$as$u20$core..future..future..Future$GT$::poll::hcc172718f0cf0e14(self=Pin<&mut core::future::from_generator::GenFuture<generator-0>> @ 0x00007fffffffb2a8, cx=0x00007fffffffb6e8) at mod.rs:80
    frame #12: 0x0000555558ed1b10 deno`_$LT$core..pin..Pin$LT$P$GT$$u20$as$u20$core..future..future..Future$GT$::poll::h509608d74c70cf83(self=Pin<&mut core::pin::Pin<alloc::boxed::Box<Future>>> @ 0x00007fffffffb308, cx=0x00007fffffffb6e8) at future.rs:119
    frame #13: 0x00005555576b9144 deno`_$LT$core..pin..Pin$LT$P$GT$$u20$as$u20$core..future..future..Future$GT$::poll::hdb0ec5b50dca608b(self=Pin<&mut core::pin::Pin<&mut core::pin::Pin<alloc::boxed::Box<Future>>>> @ 0x00007fffffffb378, cx=0x00007fffffffb6e8) at future.rs:119
    frame #14: 0x00005555577cb835 deno`tokio::runtime::basic_scheduler::Inner$LT$P$GT$::block_on::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h8042f027fee114f8 at basic_scheduler.rs:184
    frame #15: 0x0000555557aa388b deno`tokio::coop::with_budget::_$u7b$$u7b$closure$u7d$$u7d$::h7e247885c3373696(cell=0x00007ffff7fae74a) at coop.rs:121
    frame #16: 0x000055555799e5a4 deno`std::thread::local::LocalKey$LT$T$GT$::try_with::he213136d66de32de(self=0x000055555c2eef10, f=<unavailable>) at local.rs:272
    frame #17: 0x000055555799c8ee deno`std::thread::local::LocalKey$LT$T$GT$::with::hb1d9fe9af9e2c036(self=0x000055555c2eef10, f=<unavailable>) at local.rs:248
    frame #18: 0x00005555577cb04b deno`tokio::runtime::basic_scheduler::Inner$LT$P$GT$::block_on::_$u7b$$u7b$closure$u7d$$u7d$::hb1f48ffe1394e62e [inlined] tokio::coop::with_budget::h89cf0ae4e118e522(budget=(__0 = core::option::Option<u8> @ 0x00007fffffffb880), f=closure-0 @ 0x00007fffffffb888) at coop.rs:114
    frame #19: 0x00005555577cb014 deno`tokio::runtime::basic_scheduler::Inner$LT$P$GT$::block_on::_$u7b$$u7b$closure$u7d$$u7d$::hb1f48ffe1394e62e [inlined] tokio::coop::budget::h3c6c5ea873e7d1b4(f=closure-0 @ 0x00007fffffffb848) at coop.rs:98
    frame #20: 0x00005555577caf72 deno`tokio::runtime::basic_scheduler::Inner$LT$P$GT$::block_on::_$u7b$$u7b$closure$u7d$$u7d$::hb1f48ffe1394e62e(scheduler=0x00007fffffffbc90, context=0x00007fffffffb9f0) at basic_scheduler.rs:184
    frame #21: 0x00005555577cd905 deno`tokio::runtime::basic_scheduler::enter::_$u7b$$u7b$closure$u7d$$u7d$::h6c9110896abccd2c at basic_scheduler.rs:266
    frame #22: 0x0000555557926a9e deno`tokio::macros::scoped_tls::ScopedKey$LT$T$GT$::set::h508b37dddfc2a5b2(self=0x000055555c33a000, t=0x00007fffffffb9f0, f=<unavailable>) at scoped_tls.rs:61
    frame #23: 0x00005555577cd49d deno`tokio::runtime::basic_scheduler::enter::h53a83b957a37458d(scheduler=0x00007fffffffbc90, f=closure-0 @ 0x00007fffffffbb70) at basic_scheduler.rs:266
    frame #24: 0x00005555577c9c8e deno`tokio::runtime::basic_scheduler::Inner$LT$P$GT$::block_on::h4bd1938d84c8e726(self=0x00007fffffffbc90, future=Pin<&mut core::pin::Pin<alloc::boxed::Box<Future>>> @ 0x00007fffffffbbb0) at basic_scheduler.rs:176
    frame #25: 0x00005555577cbbfd deno`tokio::runtime::basic_scheduler::InnerGuard$LT$P$GT$::block_on::hbf2371d1a480703f(self=0x00007fffffffbc90, future=Pin<&mut core::pin::Pin<alloc::boxed::Box<Future>>> @ 0x00007fffffffbbf0) at basic_scheduler.rs:405
    frame #26: 0x00005555577cc3ac deno`tokio::runtime::basic_scheduler::BasicScheduler$LT$P$GT$::block_on::hac0183b0d92ddf07(self=0x00007fffffffc160, future=(pointer = core::future::future::Box<core::future::future::Future> @ 0x00007fffffffc018)) at basic_scheduler.rs:136
    frame #27: 0x000055555765e445 deno`tokio::runtime::Runtime::block_on::h8fdeeef88272dca2(self=0x00007fffffffc158, future=(pointer = core::future::future::Box<core::future::future::Future> @ 0x00007fffffffc098)) at mod.rs:450
    frame #28: 0x0000555556412cfc deno`deno::tokio_util::run_basic::h4d11dadaf01fb6ca(future=(pointer = core::future::future::Box<core::future::future::Future> @ 0x00007fffffffc148)) at tokio_util.rs:23
    frame #29: 0x0000555555bfe8f6 deno`deno::main::h7b4b3a13080cd158 at main.rs:1281
    frame #30: 0x0000555555cfa8eb deno`core::ops::function::FnOnce::call_once::hf11a4bab1df3b434((null)=(deno`deno::main::h7b4b3a13080cd158 at main.rs:1247), (null)=<unavailable>) at function.rs:227
    frame #31: 0x000055555655188e deno`std::sys_common::backtrace::__rust_begin_short_backtrace::hf62ea50fa0dc7862(f=(deno`deno::main::h7b4b3a13080cd158 at main.rs:1247)) at backtrace.rs:137
    frame #32: 0x000055555618daa1 deno`std::rt::lang_start::_$u7b$$u7b$closure$u7d$$u7d$::hebda9bd309f4b36e at rt.rs:66
    frame #33: 0x000055555a9bff87 deno`std::rt::lang_start_internal::h142b9cc66267fea1 [inlined] core::ops::function::impls::_$LT$impl$u20$core..ops..function..FnOnce$LT$A$GT$$u20$for$u20$$RF$F$GT$::call_once::he179d32a5d10d957 at function.rs:259
    frame #34: 0x000055555a9bff80 deno`std::rt::lang_start_internal::h142b9cc66267fea1 [inlined] std::panicking::try::do_call::hcb3d5e7be089b2b4 at panicking.rs:381
    frame #35: 0x000055555a9bff80 deno`std::rt::lang_start_internal::h142b9cc66267fea1 [inlined] std::panicking::try::h7ac93b0cd56fb701 at panicking.rs:345
    frame #36: 0x000055555a9bff80 deno`std::rt::lang_start_internal::h142b9cc66267fea1 [inlined] std::panic::catch_unwind::h7b40e396c93a4fcd at panic.rs:382
    frame #37: 0x000055555a9bff80 deno`std::rt::lang_start_internal::h142b9cc66267fea1 at rt.rs:51
    frame #38: 0x000055555618da77 deno`std::rt::lang_start::h7f9fdd649ebdfec2(main=(deno`deno::main::h7b4b3a13080cd158 at main.rs:1247), argc=5, argv=0x00007fffffffe088) at rt.rs:65
    frame #39: 0x0000555555bfeb7a deno`main + 42
    frame #40: 0x00007ffff77ebbf7 libc.so.6`__libc_start_main(main=(deno`main), argc=5, argv=0x00007fffffffe088, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffe078) at libc-start.c:310
    frame #41: 0x0000555555b2dd9a deno`_start + 42
(lldb)
```

**Comment by bnoordhuis** [comment]
I'm reasonably sure this is caused by a `V8InspectorClient` vtable size mismatch that isn't reflected in rusty_v8 yet. Or, since the rusty_v8 test suite is passing, maybe it's deno that does something weird.

File: runtime/inspector.rs [hunk 1]
```diff
@@ -491,54 +491,47 @@ impl DenoInspector {
...
-      }
+      let debugger_url = info.get_websocket_debugger_url();
+      server.register_inspector(info);
+      Some(debugger_url)
+    } else {
+      None
+    };
+
+    let mut out = Box::new(Self {
+      v8_inspector_client: v8::inspector::V8InspectorClientBase::new::<Self>(),
```
Comment (piscisaureus) [inline runtime/inspector.rs:511]: This moves a value that cannot be safely moved.

**Comment by ry** [comment]
closed in favor of #9180

**Pull request closed without merging**